### PR TITLE
Fix double counting of createImageBitmap requests in RequestScheduler

### PR DESCRIPTION
**Pull request by steve9164**
A potential fix to #8162.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @steve9164! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://github.com/TerriaJS/cesium/blob/fix-double-counting-upstream/CONTRIBUTORS.md) file!
* :x: Missing CLA.
    * Please send in a [Contributor License Agreement](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) and comment back here to let us know to check this!
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/fix-double-counting-upstream/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by kring** [comment]
@steve9164 is a member of my team and is covered by our CLA.

**Comment by OmarShehata** [comment]
Thanks for confirming that Kevin, we added Stephen to Data61's corporate CLA.

**Comment by hpinkos** [comment]
@steve9164 please merge in master

**Comment by steve9164** [comment]
Merged!

**Comment by hpinkos** [comment]
@tfili or @OmarShehata could you review?

File: Specs/Core/VRTheWorldTerrainProviderSpec.js [hunk 1]
```diff
@@ -244,8 +244,8 @@ describe('Core/VRTheWorldTerrainProvider', function() {
         it('provides HeightmapTerrainData', function() {
             var baseUrl = 'made/up/url';
 
-            Resource._Implementations.createImage = function(url, crossOrigin, deferred) {
-                expect(url.indexOf('.tif?cesium=true')).toBeGreaterThanOrEqualTo(0);
+            Resource._Implementations.createImage = function(request, crossOrigin, deferred) {
+                expect(request.url.indexOf('.tif?cesium=true')).toBeGreaterThanOrEqualTo(0);
 
                 // Just return any old image.
                 Resource._DefaultImplementations.createImage(imageUrl, crossOrigin, deferred);
```
Comment (tfili) [inline Specs/Core/VRTheWorldTerrainProviderSpec.js]: Shouldn't all the `Resource._DefaultImplementations.createImage` calls in the tests now take `request`? There are a bunch spread throughout the tests.

Comment (steve9164) [reply Specs/Core/VRTheWorldTerrainProviderSpec.js]: You're right. And all of the spies need to be changed too. For example this one:
https://github.com/AnalyticalGraphicsInc/cesium/blob/24aa1f7b1a2ef404cc88ebea4020521d7d945f87/Specs/Scene/UrlTemplateImageryProviderSpec.js#L113-L116

**Comment by steve9164** [comment]
I've fixed the tests and I think this is ready for another review. I think the CI error is not due to my changes, since it comes from `loadImageElement/image.onerror@Source/Core/Resource.js`, so it sounds like the image failed to load somehow before reaching my changes.

**Comment by steve9164** [comment]
Tests pass now and the new ES6 changes have been merged in. I think this is a pretty big issue. At the moment any Cesium app running on a version of Chrome that supports `createImageBitmap` will issue only 3 simultaneous requests for image tiles per domain.

**Comment by OmarShehata** [comment]
> At the moment any Cesium app running on a version of Chrome that supports createImageBitmap will issue only 3 simultaneous requests for image tiles per domain.

This is still true with this PR, it's just not counting them anymore, right?

**Comment by steve9164** [comment]
With this PR an image request will correctly be counted as 1 request (instead of 2), so Cesium's default request limit of 6 per domain will be correctly applied, instead of Cesium incorrectly limiting to 3 real requests.

Here are some screenshots from Chrome running my changes and running the current Cesium. I set Chrome to throttle requests (added a 4000ms latency so that the number of in-flight requests was obvious), used Stamen Watercolour map (because it uses the `createImageBitmap` code in Chrome) and panned around to request new tiles. (Note that stamen watercolour loads from multiple subdomains, but I'm just looking at requests to 1 subdomain, because there should be a max of 6 to each subdomain, but there isn't)

On http://localhost:8080/Apps/CesiumViewer/index.html with my changes:
image

On https://cesiumjs.org/Cesium/Apps/CesiumViewer/index.html:
image

In the first there are 6 concurrent requests (which is correct) and the second screenshot shows only 3 concurrent requests due to the double counting issue.

**Comment by OmarShehata** [comment]
Thanks for the explanation @steve9164. I wasn't thinking about how overcounting was interfering with the request scheduler's throttling. We should definitely try to get this fix in this upcoming release since it sounds like this is slowing down imagery loading.

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @steve9164! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @steve9164! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @steve9164! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by kring** [comment]
If I don't hear any complaints, I'm going to edit this lightly (e.g. sync with master) and then merge it. Probably today. It makes Cesium load imagery twice as fast. 🐎

**Comment by kring** [comment]
Thanks @steve9164, sorry this took so long to merge!

**Pull request merged**